src/server/game/AI/SmartScripts/SmartAI.cpp

```diff
@@ -352,11 +352,6 @@ bool SmartAI::IsEscortInvokerInRange()
     return true;
 }
 
-void SmartAI::WaypointStarted(uint32 nodeId, uint32 pathId)
-{
-    GetScript()->ProcessEventsFor(SMART_EVENT_WAYPOINT_START, nullptr, nodeId, pathId);
```

Comment:
and this one? also another one you cant imagine where to use it? my god

Reply:
SMART_EVENT_WAYPOINT_START was broken for years, when it was fixed, turned out it even cannot be used the way we want to use it. Then turned out Blizz uses no hooks for waypoints in DB scripts. On top of that WaypointStarted was always barely used because it's simply not necessary to use it in general

You can read about all of that here https://github.com/TrinityCore/TrinityCore/issues/25179#issuecomment-941750597 and below